client/my-sites/checkout/cart/cart-plan-ad.jsx

```diff
@@ -26,11 +30,15 @@ class CartPlanAd extends Component {
 
 	shouldDisplayAd = () => {
 		const { cart, isDomainOnly, selectedSite } = this.props;
+		const domainRegistrations = cartItems.getDomainRegistrations( cart );
+		const isDomainPremium = domainRegistrations.length === 1 && get( domainRegistrations[ 0 ], 'extra.premium', false );
 
 		return ! isDomainOnly &&
 			cart.hasLoadedFromServer &&
+			! cart.hasPendingServerUpdates &&
```

Comment:
Added this check, otherwise we got a flash of the upsell, while the updates from the server were fetched.

Reply:
Much smart. So updates. Wow!

Reply:
Seriously, this thing is good. :laughing: